android/java/org/chromium/chrome/browser/settings/BraveVpnPreferences.java

```diff
@@ -0,0 +1,483 @@
+                            }
+                        });
+                    }
+                }
+
+                @Override
+                public void onLost(Network network) {
+                    BraveVpnUtils.dismissProgressDialog();
+                    if (getActivity() != null) {
+                        BraveVpnUtils.cancelBraveVpnNotification(getActivity());
+                        getActivity().runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                if (mVpnSwitch != null)
+                                    mVpnSwitch.setChecked(
+                                            BraveVpnProfileUtils.getInstance(getActivity())
+                                                    .isVPNConnected());
+                            }
+                        });
+                    }
+                }
+            };
+
+    private void verifyPurchase(boolean isVerification) {
+        List<Purchase> purchases = InAppPurchaseWrapper.getInstance().queryPurchases();
+        if (purchases.size() == 1) {
+            Purchase purchase = purchases.get(0);
+            purchaseToken = purchase.getPurchaseToken();
+            productId = purchase.getSkus().get(0).toString();
+            if (isVerification) {
+                BraveVpnNativeWorker.getInstance().verifyPurchaseToken(
+                        purchaseToken, productId, "subscription", getActivity().getPackageName());
+            } else {
+                BraveVpnNativeWorker.getInstance().getSubscriberCredential("subscription",
+                        productId, "iap-android", purchaseToken, getActivity().getPackageName());
+            }
+        } else {
+            BraveVpnPrefUtils.setBraveVpnStringPref(
+                    BraveVpnPrefUtils.PREF_BRAVE_VPN_PURCHASE_TOKEN, "");
+            BraveVpnPrefUtils.setBraveVpnStringPref(
+                    BraveVpnPrefUtils.PREF_BRAVE_VPN_PRODUCT_ID, "");
+            BraveVpnPrefUtils.setBraveVpnStringPref(
+                    BraveVpnPrefUtils.PREF_BRAVE_VPN_PURCHASE_EXPIRY, "");
+            BraveVpnPrefUtils.setBraveVpnBooleanPref(
+                    BraveVpnPrefUtils.PREF_BRAVE_VPN_SUBSCRIPTION_PURCHASE, false);
+            if (BraveVpnProfileUtils.getInstance(getActivity()).isVPNConnected()) {
+                BraveVpnProfileUtils.getInstance(getActivity()).stopVpn();
+            }
+            BraveVpnProfileUtils.getInstance(getActivity()).deleteVpnProfile();
+            BraveVpnUtils.openBraveVpnPlansActivity(getActivity());
+        }
+    }
+
+    @Override
+    public void onVerifyPurchaseToken(String jsonResponse, boolean isSuccess) {
+        if (isSuccess) {
+            String purchaseExpiry = BraveVpnUtils.getPurchaseExpiryDate(jsonResponse);
+            if (!purchaseExpiry.isEmpty()
+                    && Long.parseLong(purchaseExpiry) >= System.currentTimeMillis()) {
```

Comment:
Maybe add try/catch block to be ready for `NumberFormatException` if the data will not be expected?